ledger/eval/eval.go

```diff
@@ -1165,12 +1241,27 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 	return nil
 }
 
+func (cs *roundCowState) takeFee(tx *transactions.Transaction, senderRewards *basics.MicroAlgos, ep *logic.EvalParams) error {
+	err := cs.Move(tx.Sender, ep.Specials.FeeSink, tx.Fee, senderRewards, nil)
+	if err != nil {
+		return err
+	}
+	// transactions from FeeSink should be exceedingly rare. But we can't count
+	// them in feesCollected because there are no net algos added to the Sink
+	if tx.Sender == ep.Specials.FeeSink {
+		return nil
+	}
+	// overflow impossible, since these sum the fees actually paid. 10B algo limit
+	cs.feesCollected, _ = basics.OAddA(cs.feesCollected, tx.Fee)
```

Comment:
`feesCollected` isn't supposed to count fees paid by the feesink (if we allowed it). I might have removed that code since it's our intent to outlaw all feesink transactions.

Reply:
The code is still there. And I'm saying, even with that restriction, if the fee sink can send algos to another account, that account can "send" then right back with a txn that has a huge fee. Repeat this process and you're double counting algos.

But the fee sink can't spend algos, so it's just a thought exercise. Resolving this now